crates/circuit/src/circuit_data.rs

```diff
@@ -1957,6 +2535,172 @@ impl CircuitData {
             _ => self.set_global_phase(add_global_phase(&self.global_phase, value)?),
         }
     }
+
+    /// Add a classical variable to the circuit.
+    ///
+    /// # Arguments:
+    ///
+    /// * var: the new variable to add.
+    /// * type_: the type the variable should have in the circuit.
+    ///
+    /// # Returns:
+    ///
+    /// The [Var] index of the variable in the circuit.
+    pub fn add_var(&mut self, var: expr::Var, type_: CircuitVarType) -> PyResult<Var> {
```

Comment:
Little nitpick there, and might not be for this PR specifically but for rust native functions we should try and stay clear of using `PyResult` for any fallible functions, we should definitely have a Rust native `CircuitError` counter.

Reply:
I completely agree, however this practice is really quite widespread currently.